halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/spinnaker/v1/profile/deck/DeckProfileFactory.java

```diff
@@ -72,7 +74,7 @@ protected void setProfile(
       Profile profile,
       DeploymentConfiguration deploymentConfiguration,
       SpinnakerRuntimeSettings endpoints) {
-    StringResource configTemplate = new StringResource(profile.getBaseContents());
+    ObjectResource configTemplate = new ObjectResource(profile.getBaseContents());
```

Comment:
`profile.getBaseContents()` works a bit different than the java services, in Deck's case, it downloads a settings.js template whose values are replaced via the StringResource. Since we do not control the settings.js template format, we'll probably have to do a string replace to add the plugins key and value, where the value should be the format given in ObjectReplaceTemplateResource